### PR TITLE
creating ubi9-py39 overlay

### PR DESCRIPTION
**Pull request by Gregory-Pereira**
## Related Issues and Dependencies

Addresses: https://github.com/thoth-station/s2i-thoth/issues/234

## This introduces a breaking change

- [ ] Yes
- [x] No

## This Pull Request implements

Adding a new ubi9 solver image.

## Description

Not entirely sure about the s2i-patch, going to test it.

/cc @harshad16

### CODE REVIEW

File: ubi9-py39/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,51 @@
+# Thoth's extension to OpenShift's S2I build
+FROM registry.access.redhat.com/ubi9/python-39:1-59
```
Comment (goern) [inline ubi9-py39/Dockerfile:2]: could you use registry.access.redhat.com/ubi9/python-39@sha256:6329d5779e89630928d5fa65a7b7b824f6bf41357f5db1688f0efafcbe48b14d

File: ubi9-py39/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+
+ENV SUMMARY="Thoth's Source-to-Image for Python ${PYTHON_VERSION} applications" \
+    DESCRIPTION="Thoth's Source-to-Image for Python ${PYTHON_VERSION} applications. This toolchain is based on Red Hat UBI9. It includes pipenv." \
+    THOTH_S2I_NAME=quay.io/thoth-station/s2i-thoth-ubi9-py39 \
+    THOTH_S2I_VERSION=0.34.1 \
```
Comment (goern) [inline ubi9-py39/Dockerfile:7]: @harshad16 I wonder if we should remove these labels?!

**Review by goern — CHANGES_REQUESTED** [review]

File: ubi9-py39/Dockerfile (hunk 2 above)
Comment (harshad16) [reply ubi9-py39/Dockerfile:7]: These env vars are used down the line , for thoth resolver notice which version of thoth base image were used.

Comment (harshad16) [reply ubi9-py39/Dockerfile:7]: @Gregory-Pereira  can you update this version to `v0.35.0`

File: ubi9-py39/Dockerfile [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+    authoritative-source-url="https://quay.io/thoth-station/s2i-thoth" \
+    version="$THOTH_S2I_VERSION" \
+    release="0" \
+    ninja.thoth-station.version="0.7.0-dev" \
+    maintainer="Thoth Station <aicoe-thoth@redhat.com>"
+
+USER 0
+COPY ./s2i_assemble.patch /tmp/s2i_assemble.patch
+COPY ./requirements.txt $HOME/requirements.txt
+RUN TMPFILE=$(mktemp) && \
```
Comment (goern) [inline ubi9-py39/Dockerfile:30]: it looks like this should be one RUN command with && \ on each line?! @Gregory-Pereira

**Comment by goern** [comment]
a root cause for this to fail build seems to be 
```
Collecting requests==2.28.1
  Downloading requests-2.28.1-py3-none-any.whl (62 kB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 62.8/62.8 kB 1.4 MB/s eta 0:00:00
Installing collected packages: requests
  Attempting uninstall: requests
    Found existing installation: requests 2.25.1
ERROR: Cannot uninstall requests 2.25.1, RECORD file not found. Hint: The package was installed by rpm.
09-06-22 14:04:55 micropipenv  ERROR  Failed to install requirements, dependency 'requests' could not be installed
```

File: ubi9-py39/Dockerfile [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+    TMPFILE_ASSEMBLE=$(mktemp) && \
+    pushd "${STI_SCRIPTS_PATH}" && \
+    ls /tmp/s2i_assemble.patch && \
+    patch -p 1 </tmp/s2i_assemble.patch && popd && \
```
Comment (harshad16) [inline ubi9-py39/Dockerfile:34]:
```suggestion
    pushd "${STI_SCRIPTS_PATH}" && patch -p 1 </tmp/s2i_assemble.patch && popd && \
```

lets remove the ls from this.

File: ubi9-py39/Dockerfile [hunk 5]
```diff
@@ -0,0 +1,49 @@
+# Thoth's extension to OpenShift's S2I build
+FROM registry.access.redhat.com/ubi9/python-39@sha256:6329d5779e89630928d5fa65a7b7b824f6bf41357f5db1688f0efafcbe48b14d
+
+ENV SUMMARY="Thoth's Source-to-Image for Python ${PYTHON_VERSION} applications" \
+    DESCRIPTION="Thoth's Source-to-Image for Python ${PYTHON_VERSION} applications. This toolchain is based on Red Hat UBI9. It includes pipenv." \
+    THOTH_S2I_NAME=quay.io/thoth-station/s2i-thoth-ubi9-py39 \
+    THOTH_S2I_VERSION=0.34.1 \
```
Comment (harshad16) [inline ubi9-py39/Dockerfile:7]:
```suggestion
    THOTH_S2I_VERSION=0.35.0 \
```

**Review by harshad16 — CHANGES_REQUESTED** [review]
all looks good now
thanks @goern 

few last change is needed and we can merge this.

**Review by harshad16 — APPROVED** [review]
/lgtm
/approve
Thanks for the work.

**Pull request merged**